Src/BootCamp.Chapter1/ArrayOperations.cs

```diff
@@ -88,8 +174,43 @@ public static int[] InsertLast(int[] array, int number)
         /// <returns>A new array with element inserted at a given index. If an array is empty or null, returns new array with number in it.</returns>
         public static int[] InsertAt(int[] array, int number, int index)
         {
-            // ToDo: implement.
-            return array;
+            var resultArray = InsertElementInArray(array, number, index);
+            return resultArray;
+        }
+
+        private static int[] InsertElementInArray(int[] array, int number, int index)
+        {
+            if (array == null || array.Length == 0)
+            {
+                if (index == 0)
+                {
+                    var newArrayNumber = new[] { number };
+                    return newArrayNumber;
+                }
+            }
+
+            var arrayLength = array.Length;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['array' is null on at least one execution path.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5011611)